src/machine_upgrade.py

```diff
@@ -11,22 +11,22 @@
 import time
 import typing
 
-import charms.opensearch.v0.constants_charm as constants_charm
 import ops
 
 import upgrade
-from opensearch import OpenSearchSnap
+from config import Config
 
 logger = logging.getLogger(__name__)
 
-_SNAP_REVISION = str(constants_charm.OPENSEARCH_SNAP_REVISION)
+_SNAP_REVISION = str(Config.SNAP_PACKAGES[0][3])
```

Comment:
```suggestion
_SNAP_REVISION = str(Config.SNAP_PACKAGES[0][2])
```
think this might be off by 1